README.md

```diff
@@ -1,189 +1,159 @@
+                  toHoliday:(NSString *)holiday
+                   inSeason:(NSString *)season
+                 inDatabase:(NSDictionary *)database;
+                 
 @end
 ```
 
-Having defined this `_size` variable, we can now set its value and get its value throughout the class!
-
-Keep in mind: Instance variables with protected scope are held in memory until the entire instance of the object is released from memory. For instance, a variable that holds a `UIView` would be held in memory until the `UIViewController` class it is instantiated within has also been released from memory.
+It would have been much easier to write those methods to access the dictionary as a property belonging to `FISAppDelegate`:
 
+```objc
+// FISAppDelegate.h
 
+@interface FISAppDelegate : UIResponder <UIApplicationDelegate>
 
-###Public instance variables 
-#####(or instance variables with "global" scope)
-
-So what if we want to access this variable from outside of the class? What if we want to set its state when something changes in another class? We can do that with "accessor" methods.
-
-There are two standard types of accessor methods: "Getters" and "Setters". Let's take a look at each.
+@property (strong, nonatomic) NSDictionary *database;
 
+- (NSArray *)holidaysInSeason:(NSString *)season;
 
+- (NSArray *)suppliesInHoliday:(NSString *)holiday
+                      inSeason:(NSString *)season;
 
-###Getter
+- (BOOL)holiday:(NSString* )holiday
+     isInSeason:(NSString *)season;
 
-In order to get the value of the `_size` instance variable from outside the class `Product`, we must declare and define a method that "gets" that value. We call this method, quite appropriately, the "getter" method. Here is an example of a default getter for the `NSNumber` instance variable called `_size`. 
+- (BOOL)supply:(NSString *)supply
+   isInHoliday:(NSString *)holiday
+      inSeason:(NSString *)season;
 
-######Example
-```objc
-//Product.m
+- (void)addHoliday:(NSString *)holiday
+          toSeason:(NSString *)season;
 
--(NSNumber *)size {
-    return _size;
-}
+- (void)addSupply:(NSString *)supply
+        toHoliday:(NSString *)holiday
+         inSeason:(NSString *)season;
 
+@end 
 ```
+Each of the methods could have accessed the `database` property from within their own implementations. Also, the test file would not have needed to be responsible for the `database` dictionary and the final two methods would not have needed to provide a return since the `database` dictionary would have been owned as a property by the `FISAppDelegate` instance.
 
+**Top-tip:** *It's standard practice to group all of the property declarations together and write them above the method declarations. Don't mix your property declarations with your method declarations.*
 
+## Declaring Properties
 
-###Setter
+A property declaration can only be placed inside the `@interface` section of a class file. In most cases you will use the following syntax:
```

Comment:
I left that out intentionally because `.m` files can have an `@interface` section also, which I'm working on discussing in the fifth reading about Visibilty (Public/Private). I don't want the curriculum to contradict itself.